src/jit/gentree.h

```diff
@@ -4369,20 +4435,20 @@ struct GenTreePutArgStk : public GenTreeUnOp
 
     GenTreePutArgStk(genTreeOps oper,
                      var_types  type,
-                     unsigned slotNum FEATURE_UNIX_AMD64_STRUCT_PASSING_ONLY_ARG(unsigned numSlots)
-                         FEATURE_UNIX_AMD64_STRUCT_PASSING_ONLY_ARG(bool isStruct),
+                     unsigned slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(unsigned numSlots)
+                         PUT_STRUCT_ARG_STK_ONLY_ARG(bool isStruct),
```

Comment:
#if FEATURE_FASTTAILCALL has added one additional field and one addtional arg to constructor.  But that has added unnecessary #ifdef clutter here and as well in lower.cpp (in call lowering logic)
We should have just added param "_putInIncomingArgArea" with default value of false towards the end.  This param would be simply not passed or ignored on targets where FEATURE_FASTTAILCALL is not set. That way it would have been much cleaner.

Couldn't resist myself making this comment, though I know it wasn't introduced by you.
Leave it up to you.